frontend/src/DeleteBook.tsx

```diff
@@ -0,0 +1,48 @@
+import React, {useState} from 'react';
+import {BookModel} from "./BookModel";
+import axios from "axios";
+import Modal from "react-modal";
+
+type DeleteBookProps = {
+    book: BookModel;
+    reloadAllBooks: () => void;
+
+}
+
+function DeleteBook(props: DeleteBookProps) {
+    const [modalIsOpen, setModalIsOpen] = useState<boolean>(false)
+    const deleteBook = (id: string) => {
+        axios.delete("/api/books/" + id)
+            .then(props.reloadAllBooks)
+            .then(() => alert("Remove book with id " + props.book.id + "successful!"))
+            .catch(error => console.error("DELETE Error: " + error))
+    }
+    const openModal = () => {
+        setModalIsOpen(true)
+    }
+
+    const closeModal = () => {
+        setModalIsOpen(false)
+    }
+
+    return <>
+        <button type={"submit"} onClick={openModal}>Delete</button>
```

Comment:
you use here an type submit but the button isnt within an form so change the type to button :)